### PR TITLE
Allow 1d and 2d normal fields in distributed mode

### PR DESCRIPTION
**Pull request by ASKabalan**
This pull request introduces improvements to the handling of distributed array shapes and sharding specifications in `jaxpm/distributed.py`. The main focus is on making sharding logic more robust and flexible, especially when working with arrays of varying dimensionality. The most important changes are grouped below:

Enhancements to sharding and shape handling:

* Improved `get_local_shape` to more accurately compute local shapes by mapping axis names to device sizes and handling cases where the mesh or sharding spec does not match the array dimensionality.
* Added `get_sharding_for_shape`, a new helper function that trims the sharding specification to match the dimensionality of the target array, ensuring correct sharding for lower-dimensional arrays.
* Updated `normal_field` to call `get_sharding_for_shape` before using sharding, ensuring the sharding spec matches the array shape.

API and import improvements:

* Extended imports to include `NamedSharding` from `jax.sharding`, supporting the new sharding logic.

### CODE REVIEW

File: tests/test_distributed_pm.py [hunk 1]
```diff
@@ -92,7 +92,8 @@ def test_distrubted_pm(simulation_config, initial_conditions, cosmo, order,
     print("Done with single device run")
     # MULTI DEVICE RUN
```
Comment (ASKabalan) [inline tests/test_distributed_pm.py:94]: These are related to the Auto / Explicit mesh axis

I don't support them https://jaxdecomp.readthedocs.io/en/latest/07-xla_sharding_configuration.html#auto-vs-explicit-axis-types
Because JAX does not provide a way to define rules for them

no fix yet from JAX https://github.com/jax-ml/jax/issues/34811
